packages/lsp/src/connection.ts

```diff
@@ -268,16 +268,25 @@ export class LSPConnection extends LspWsConnection implements ILSPConnection {
     return this._serverInitialized;
   }
 
+  /**
+   * A signal emitted when the connection is disposed.
+   */
+  get disposed(): ISignal<this, void> {
+    return this._disposed;
+  }
+
   /**
    * Dispose the connection.
    */
   dispose(): void {
     if (this._isDisposed) {
```

Comment:
You should switch `_isDisposed` to private and use the public getter `isDisposed` here.